src/app/clusters/mode-base-server/readme.md

```diff
@@ -0,0 +1,35 @@
+# Mode Base and its aliases
+
+ModeBase is a pseudo cluster. It has no cluster ID. It exists only to be derived
+into other clusters.
+
+# How to use a ModeBase derived cluster
+
+To use a ModeBase derived cluster, you need to
+
+-   create a class that inherits the `ModeBase::Instance` class. For examples
+    look at the `<alias name>-mode.x` files in the `src` and `include`
+    directorins in `examples/all-clusters-app/all-clusters-common`.
+-   For this class implement the `NumberOfModes`, `GetModeLabelByIndex`,
+    `GetModeValueByIndex` `GetModeTagsByIndex` and `HandleChangeToMode`
+    functions. Look at the documentation in `mode-base-server.h` for more
+    information about these functions. Cluster specific enums can be accessed
+    from the cluster namespace.
+-   Optionally implement the `AppInit` function.
+-   In your main file, instantiate your `ModeBase::Instance` inherited class.
+    See the constructor documentation in `mode-base-server.h`.
+-   Call the `.Init()` function of your instance.
+
+**Note** Zap accessor functions for these clusters will not return true values.
+Use the instance's `Update...` and `Get...` functions to access the attributes.
+
+# How to add new derived clusters
+
+Once a ModeBase derived cluster has been defined in the spec, adding the
+implementation to ModeBase using the following steps
+
+1.  Translate the spec as an XML in `src/app/zap-templates/zcl/data-model/chip`.
+    You can look at similar files on how to do this.
+2.  Add the cluster ID to `AliasedClusters` in `mode-base-cluster-objects.h`.
```

Comment:
Is it needed to have an exhaustive list here? Could this be runtime determined via registration?

Reply:
I don't know how the registration process atm can know which clusters are aliases of ModeBase. So we have to keep a list somewhere. Eventually we can have zap generate this list from the XMLs. This list is used during the init process to ensure that the user has given a valid ID to the instance constructor. That is an ID that belongs to a ModeBase aliased cluster.

Reply:
Why do we care that it's "a valid ID" ? Why do we have to do this additional enforcement here? Suggest removing the constraint

Reply:
Done [here](https://github.com/project-chip/connectedhomeip/pull/27504/commits/c41b94e0bf810051bf2e73c157ab1ef0793e2b44).